### DISCUSSION TITLE
How to create generator from compiler object?

### DISCUSSION THREAD

**Discussion by stsp** [mesonbuild/meson#12563] [Q&A] [answered]
I want to define the generator that uses
the C preprocessor. So I do this:
```
incdir = include_directories(rel_inc)
cpp = meson.get_compiler('c')
sgen = generator(cpp.oops?
```

Generator requires the "Executable", "ExternalProgram"
objects, so how can I get one from
the  compiler object? And where should
I put this `incdir`?

**Comment by QuLogic** [comment, marked as answer]
Why do you need a generator instead of calling the [`preprocess` method](https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerpreprocess)?

**Reply by stsp** [reply]
Thanks, that seems to work.
